### ISSUE TITLE
 InteractiveViewer position shift after resizing with boundaryMargin

### ISSUE THREAD

**Issue by Elvincth** [flutter/flutter#109142]
After adding boundaryMargin and `constrained: false`, when resizing the InteractiveViewer the child will shift its position. 

## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Trigger a resize on the InteractiveViewer 
3. The position shifted 

**Expected results:** 
The shifted position (center) should be rendered as the default

**Actual results:** 

https://user-images.githubusercontent.com/6602723/183402113-1fc8751b-0624-494a-b636-68f46f2ec97c.mov

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black,
      body: InteractiveViewer(
       minScale: 0.1,
        maxScale: 2,
        constrained: false,
        boundaryMargin: const EdgeInsets.all(200),
        child: Container(
          width: 990,
          height: 600,
          color: Colors.red
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5 21G72 darwin-arm, locale en-US)
    • Flutter version 3.0.5 at /opt/homebrew/Caskroom/flutter/3.0.5/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (4 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/chutsunhim/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/chutsunhim/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.70.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • iPad Pro (12.9-inch) (5th generation) (mobile) • ED135AFF-8411-4F15-A9A1-B7F1D1DD2740 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)                                • macos                                • darwin-arm64   • macOS 12.5 21G72 darwin-arm
    • Chrome (web)                                   • chrome                               • web-javascript • Google Chrome 104.0.5112.79

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

</details>

**Comment by huycozy** [comment]
Hi @Elvincth 
Could you try this modified sample code and see if it's your expected behavior?

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

const Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        scaffoldBackgroundColor: darkBlue,
      ),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black,
      body: Center(
        child: InteractiveViewer(
          minScale: 0.1,
          maxScale: 2.0,
          boundaryMargin: const EdgeInsets.all(double.infinity),
          constrained: false,
          child: Container(width: 990, height: 600, color: Colors.red),
        ),
      ),
    );
  }
}

```
</details>

<details>
<summary>Demo</summary>

https://user-images.githubusercontent.com/104349824/183584579-0ad3b92b-4b52-4ae3-a940-9f515055a8db.mov

</details>

**Comment by Elvincth** [comment]
> [huycozy](/huycozy)

Hey huycozy,
Thanks for the code example. But that's not what I want, the reason I wanted to set the border margin to 200 was to limit the amount the user could pan (so they don't get too far away from the object) and it would put the object in the center of the screen. But setting the border margin to infinity doesn't achieve the effect I want.

**Comment by Elvincth** [comment]
I am able to slove it by the following code.
```dart
             double deviceWidth = MediaQuery.of(context).size.width;
                        double deviceHeight =
                            MediaQuery.of(context).size.height;
                        double scale = _controller.value.getMaxScaleOnAxis();
                        double canvasWidth = 990 * scale;
                        double canvasHeight = 600 * scale;

                        _controller.value.setTranslationRaw(
                            (deviceWidth / 2 - canvasWidth / 2),
                            (deviceHeight / 2 - canvasHeight / 2),
                            0);

                        setState(() {});
```

**Comment by huycozy** [comment]
@Elvincth Glad to see you solved the issue. I see there's a closed issue #64604 that may be related to this issue. And please look at https://github.com/flutter/flutter/issues/64604#issuecomment-680966024 that leads to the open issue #58997

**Comment by Elvincth** [comment]
> @Elvincth Glad to see you solved the issue. I see there's a closed issue #64604 that may be related to this issue. And please look at [#64604 (comment)](https://github.com/flutter/flutter/issues/64604#issuecomment-680966024) that leads to the open issue #58997

Yes very similar, I think I could share my code later on.
